### ISSUE TITLE
Alarm & Decryptor

### ISSUE THREAD

**Issue by Dankman2048** [Anuken/Mindustry-Suggestions#2731]
An Alarm. It needs power to work. When it detects an enemy mech, whether it be PvP or an AI, it notifies the player, similarly how the player gets notified that the core is under attack. A Decryptor runs on phase fabric and power, and has a chance to hack into and control an enemy base's turret within a radius.

These are ideas for 7.0, I guess

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
2. - [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
3. - [X] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
Who needs an alarm when logic is everything.

**Comment by itcannotbe** [comment]
Decryptor be like: All your meltdowns are mine

TL;DR:
OP, Too OP.

**Comment by Dankman2048** [comment]
Like I said, the Decryptor only works in a radius, preferably smaller than
the Meltdown. It also takes time to hack into turrets. I'm sorry for making
bad suggestions if that's what you want me to say.
TL,DR: bruh

**Comment by oinkymoo** [comment]
1. these should be separate suggestions
2. @itcannotbe the problem with logic-based alarms is that they have to send a signal upwards of 100 blocks, if not several hundred, and either send the signal to a display of some kind which the player may not be near enough to notice, or like, set off an explosion next to the core, which like, why would you do that
3. i agree that the decryptor is probably way too overpowered, but the limitations you propose make it nearly useless against anything but the weakest turrets

**Comment by itcannotbe** [comment]
@oinkymoo Why would you need to rely on displays when you could make some unit do something that's noticable. Like make some random unit circle you to indicate some alarm being triggered.

**Comment by lokyinng** [comment]
> 1. these should be separate suggestions
> 2. @itcannotbe the problem with logic-based alarms is that they have to send a signal upwards of 100 blocks, if not several hundred, and either send the signal to a display of some kind which the player may not be near enough to notice, or like, set off an explosion next to the core, which like, why would you do that
> 3. I agree that the decryptor is probably way too overpowered, but the limitations you propose make it nearly useless against anything but the weakest turrets

I gonna hack all your meltdown with use the power of decryptor and this plastanium wall I have found on the ground

**Comment by abrasmag3** [comment]
@oinkymoo signals can be wireless by getblocking a memory cell. It only requires one unit to do connect, and after connecting the unit can go back to default ai
